### PR TITLE
syncer: don't skip jobs from same event when comparing table checkpoint

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
close https://github.com/pingcap/dm/issues/1751

after `all_mode` paused, we could see

```sql
MySQL [all_mode]> select * from t2;
+----+-------+---------------------+
| id | name  | ts                  |
+----+-------+---------------------+
|  1 | Arya  | 2021-06-05 18:01:52 |
|  2 | Bran  | 2021-05-11 10:01:05 |
|  3 | Sansa | 2021-06-05 18:01:52 |
| 10 | 10    | 2021-06-05 18:02:02 |
| 20 | 20    | 2021-06-05 18:02:02 |
| 30 | 30    | 2021-06-05 18:02:02 |
+----+-------+---------------------+
6 rows in set (0.001 sec)

MySQL [all_mode]> select * from t1;
+----+---------+---------------------+---------------------+
| id | name    | dt                  | ts                  |
+----+---------+---------------------+---------------------+
|  1 | ar"ya   | 2021-06-05 18:01:52 | 2021-06-05 18:01:52 |
|  2 | catelyn | 2021-05-11 10:01:05 | 2021-05-11 10:01:05 |
|  3 | lalala  | NULL                | 2021-06-05 18:01:52 |
| 10 | 10      | NULL                | 2021-06-05 18:02:01 |
| 30 | 30      | NULL                | 2021-06-05 18:02:02 |
+----+---------+---------------------+---------------------+
5 rows in set (0.001 sec)
```

t1 losts a row with id = 20 of https://github.com/pingcap/dm/pull/1752/files#diff-5cc7189b51f6219ea0ce1c9920f92f84cee41b3fee093894a72ca5ff1c615b4aR2

### Check List 

Tests 

 - Integration test

Code changes

 - Has interface methods change

Related changes

 - Need to cherry-pick to the release branch
 - Need to be included in the release note

### CODE REVIEW

**Review by GMHDBJD — APPROVED** [review]
LGTM

File: syncer/checkpoint.go [hunk 1]
```diff
@@ -443,34 +443,32 @@ func (cp *RemoteCheckPoint) DeleteSchemaPoint(tctx *tcontext.Context, sourceSche
...
+//   - Query event e1, location is gset1
+//   - Rows event e2, location is gset1
 //   - XID event, location is gset2
-//   after syncer handle query event, will save table point with gset1, and when handle rows event, will compare the rows's location with table checkpoint's location in `IsNewerTablePoint`, and these two location have same gset, so we should use `>=` to compare location in this case.
-// - when enable GTID and binlog is DDL, different DDL have different GTID set, so if GTID set is euqal, it is a old table point, should use `>` to compare location in this case.
-// - when not enable GTID, just compare the position, and only when grater than the old point is newer table point, should use `>` to compare location is this case.
-func (cp *RemoteCheckPoint) IsNewerTablePoint(sourceSchema, sourceTable string, location binlog.Location, gte bool) bool {
+// We should note that e1 is not older than e2
+// For binlog position replication, currently DM will split rows changes of an event to jobs, so some job may has save position.
+// if useLE is true, we use less than or equal.
```
Comment (Ehco1996) [inline syncer/checkpoint.go:453]:
```suggestion
// if useLTE is true, we use less than or equal.
```

File: syncer/checkpoint.go [hunk 2]
```diff
@@ -443,34 +443,32 @@ func (cp *RemoteCheckPoint) DeleteSchemaPoint(tctx *tcontext.Context, sourceSche
...
+func (cp *RemoteCheckPoint) IsOlderThanTablePoint(sourceSchema, sourceTable string, location binlog.Location, useLE bool) bool {
```
Comment (Ehco1996) [inline syncer/checkpoint.go:454]:
```suggestion
func (cp *RemoteCheckPoint) IsOlderThanTablePoint(sourceSchema, sourceTable string, location binlog.Location, useLTE bool) bool {
```

File: syncer/checkpoint.go [hunk 3]
```diff
@@ -443,34 +443,32 @@ func (cp *RemoteCheckPoint) DeleteSchemaPoint(tctx *tcontext.Context, sourceSche
...
 	if !ok {
-		return true
+		return false
 	}
 	oldLocation := point.MySQLLocation()
 	cp.logCtx.L().Debug("compare table location whether is newer", zap.Stringer("location", location), zap.Stringer("old location", oldLocation))
 
-	if gte {
-		return binlog.CompareLocation(location, oldLocation, cp.cfg.EnableGTID) >= 0
+	if useLE {
```
Comment (Ehco1996) [inline syncer/checkpoint.go:468]:
```suggestion
	if useLTE {
```

**Review by Ehco1996 — COMMENTED** [review]
rest LGTM

File: syncer/checkpoint.go (hunk 1 above)
Comment (lance6716) [reply syncer/checkpoint.go:453]: most language use `LE` not `LTE` (seems a 4G terminology)

Comment (lance6716) [reply syncer/checkpoint.go:453]: https://en.wikipedia.org/wiki/Less-than_sign#Less-than_sign_plus_equals_sign

**Review by Ehco1996 — APPROVED** [review]

**Review by glorv — APPROVED** [review]

**Pull request merged**